service/src/main/java/bio/terra/workspace/app/controller/ControllerBase.java

```diff
@@ -143,12 +143,13 @@ public PrivateUserRole computePrivateUserRole(
                       userRequest),
               "validate private user is workspace member");
 
-          // Translate the incoming role list into our internal model form
+          // Translate the incoming role into our internal model form
           // This also validates that the incoming API model values are correct.
-          List<ControlledResourceIamRole> roles =
-              commonFields.getPrivateResourceUser().getPrivateResourceIamRoles().stream()
-                  .map(ControlledResourceIamRole::fromApiModel)
-                  .collect(Collectors.toList());
+          List<ControlledResourceIamRole> roles = new ArrayList<ControlledResourceIamRole>();
+          roles.add(
+              ControlledResourceIamRole.fromApiModel(
+                  commonFields.getPrivateResourceUser().getPrivateResourceIamRole()));
```

Comment:
why do you need a list here? i thought you changed it to singular now?

Reply:
I left it as a list since the code that follows expects a list. It would probably be good to refactor that portion too. I'll do that and send out an update.